test/unit/test_contraction.py

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env python
+import unittest
+from tinygrad.lazy import get_contraction
+
+class TestFlopCounter(unittest.TestCase):
+  def test_contraction(self):
+    r = get_contraction((1,2,3,4), (2,3,4))
+    self.assertEqual(r, [[0, 1], [2], [3]])
+    
+    r = get_contraction((1,2,3,1,4), (1,2,3,4))
+    self.assertEqual(r, [[0], [1], [2], [3, 4]])
+
+    r = get_contraction((1,2,3,1,4,1,1), (2,3,4))
+    self.assertEqual(r, [[0, 1], [2], [3, 4, 5, 6]])
+
+    r = get_contraction((1,2,3,4), (1,2,3*4))
+    self.assertEqual(r, [[0], [1], [2, 3]])
+
+    r = get_contraction((1,2,3,4), (2,1,3,4))
```

Comment:
These two tests bring up something very interesting.

If the new dimensions are just 1s, I could see this being

[0,1], [], [2], [3] and working to push the permute.